gobblin-config-management/gobblin-config-core/src/main/java/gobblin/config/store/hdfs/SimpleHDFSConfigStore.java

```diff
@@ -0,0 +1,374 @@
+ * {@link SimpleHDFSConfigStoreFactory}.
+ *
+ * <p>
+ *   The class assumes a directory called {@link #CONFIG_STORE_NAME} is under the root directory. This folder should
+ *   contain a directory for each version deployed to the {@link ConfigStore}. An example directory structure could look
+ *   like: <br>
+ *   <blockquote>
+ *     <code>
+ *       /root<br>
+ *       &emsp;/my-simple-store<br>
+ *       &emsp;&emsp;/_CONFIG_STORE<br>
+ *       &emsp;&emsp;&emsp;/v1.0<br>
+ *       &emsp;&emsp;&emsp;&emsp;/dataset1<br>
+ *       &emsp;&emsp;&emsp;&emsp;&emsp;/child-dataset<br>
+ *       &emsp;&emsp;&emsp;&emsp;&emsp;&emsp;/main.conf<br>
+ *       &emsp;&emsp;&emsp;&emsp;&emsp;&emsp;/includes.conf<br>
+ *       &emsp;&emsp;&emsp;&emsp;/dataset2<br>
+ *       &emsp;&emsp;&emsp;&emsp;&emsp;/main.conf<br>
+ *       &emsp;&emsp;&emsp;&emsp;&emsp;/child-dataset<br>
+ *       &emsp;&emsp;&emsp;&emsp;&emsp;&emsp;/main.conf<br>
+ *     </code>
+ *   </blockquote>
+ * </p>
+ *
+ * <p>
+ *   In the above example, the root of the store is {@code /root/my-simple-store/}. The code automatically assumes that
+ *   this folder contains a directory named {@link #CONFIG_STORE_NAME}. In order to access the dataset
+ *   {@code dataset1/child-dataset} using {@link ConfigClient#getConfig(URI)}, the specified {@link URI} should be
+ *   {@code simple-hdfs://[authority]:[port]/root/my-simple-store/dataset1/child-dataset/}. Note this is the fully
+ *   qualified path to the actual {@code main.conf} file on HDFS, with the {@link #CONFIG_STORE_NAME} and the
+ *   {@code version} directories removed.
+ * </p>
+ *
+ * <p>
+ *   All the {@link Config}s for a dataset should be put in the associated {@code main.conf} file, and all the imports
+ *   should be put in the associated {@code includes.conf} file.
+ * </p>
+ *
+ * <p>
+ *   This class is not responsible for deploying configurations from an external source to HDFS, only for reading them.
+ * </p>
+ *
+ * @see {@link SimpleHDFSConfigStoreFactory}
+ */
+public class SimpleHDFSConfigStore implements ConfigStore {
+
+  public static final String SIMPLE_HDFS_SCHEME_PREFIX = "simple-";
+  public static final String CONFIG_STORE_NAME = "_CONFIG_STORE";
+
+  private final FileSystem fs;
+  private final URI physicalStoreRoot;
+  private final Cache<String, Path> versions;
+
+  /**
+   * Constructs a {@link SimpleHDFSConfigStore} using a given {@link FileSystem} and a {@link URI} that points to the
+   * physical location of the store root.
+   *
+   * @param fs the {@link FileSystem} the {@link ConfigStore} is stored on.
+   * @param physicalStoreRoot the fully qualified {@link URI} of the store root, the {@link URI#getScheme()} of the
+   *                          {@link URI} should match the {@link FileSystem#getScheme()} of the given {@link FileSystem}.
+   */
+  protected SimpleHDFSConfigStore(FileSystem fs, URI physicalStoreRoot) {
```

Comment:
Prefer to not have the FileSystem as parameter as through physicalStoreRoot can get the FileSystem.

Also, prefer to have "URI physical_root, URI logic_root" as parameter  , then getStoreURI will return logic_root